src/main/scripts/setup/setup_helpers.py

```diff
@@ -0,0 +1,228 @@
+from fabric.api import *
+
+project="festive-courier-755"
+zone="us-central1-f"
+credentials = GoogleCredentials.get_application_default()
+compute = discovery.build('compute', 'v1', credentials=credentials)
+
+def get_email_id():
+  f = open(os.environ['HOME'] + "/.aurora.conf")
+  for line in f:
+    if 'emailid' in line:
+      return line.rstrip('\n').split("emailid=",1)[1]
+
+# Function to get mesos instances ips as a list.
+# IPs have already been written in a file.
+def get_mesos_x_ips(setup_ips_dir, x="all"):
+  if x == "masters":
+    file_path_name = setup_ips_dir + '/mesos_masters_ips'
+  elif x == "slaves":
+    file_path_name = setup_ips_dir + '/mesos_slaves_ips'
+  elif x == "all":
+    file_path_name = setup_ips_dir + '/mesos_all_ips'
+
+  try:
+    f = open(file_path_name)
+    ips = [line.rstrip('\n') for line in f]
+    f.close
+  except:
+    print ("WARN: Perhaps file %s does not exist" % file_path_name)
+    return
+  return ips
+
+# Get all IP addresses (both masters and slaves)
+def get_mesos_all_ips(setup_ips_dir):
+  return get_mesos_x_ips(setup_ips_dir, x="all")
+
+# Get master IP addresses
+def get_mesos_masters_ips(setup_ips_dir):
+  return get_mesos_x_ips(setup_ips_dir, x="masters")
+
+# Get Slaves IPs
+def get_mesos_slaves_ips(setup_ips_dir):
+  return get_mesos_x_ips(setup_ips_dir, x="slaves")
+
+# Function to get gcloud instances ips.
+# It is only for GCE.
+def get_master_instances_ips():
+  email_id = get_email_id()
+  filt = "name eq " + email_id + "-master.*"
+  results = compute.instances().list(project=project, zone=zone, filter=filt).execute()
+  ips = list()
+  for instance in results['items']:
+    ips.append(instance["networkInterfaces"][0]["networkIP"])
+  return ips
+
+def run_command(command):
+    p = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
+    (output, err) = p.communicate()
+    return output
+
+def get_instance_tag(ip):
+  command = "gcloud compute instances list | grep -w " + ip
+  output = run_command(command)     # tahir-slave1-slave-set1-0 us-central1-f n1-standard-4 10.10.0.28 RUNNING
+  instance_name = output.split()[0] # tahir-slave1-slave-set1-0
+  tag_lst = instance_name.split("-")[2:-1]    # ['slave', 'set1']
+  tag = "-".join(tag_lst)
+  #print (tag)
+  return tag
+
+def get_slave_instances_ips():
+  email_id = get_email_id()
+  filt = "name eq " + email_id + "-slave.*"
+  results = compute.instances().list(project=project, zone=zone, filter=filt).execute()
+  ips = list()
+  for instance in results['items']:
+    ips.append(instance["networkInterfaces"][0]["networkIP"])
+  return ips
+
+def spawn_instance(instance_name, os_name, machine_type="n1-standard-4", dst_user="plumgrid"):
+  instance_name = email_id + "-" + instance_name # Prefix emailid before instance name.
+  #(fd, pathname) = mkstemp(prefix="gce_key_")
+  pathname="/tmp/gce_key.txt"
+  tfile = open(pathname, 'w')
+  #tfile = os.fdopen(fd, "w")
+
+  with open("/home/muneeb/.ssh/id_rsa.pub") as f:
```

Comment:
lol ... I will clean this up. This function is not currently being used. We will remove aurora dependency by using this function.